pulsar-broker/src/main/java/org/apache/pulsar/broker/service/SystemTopicBasedTopicPoliciesService.java

```diff
@@ -362,13 +362,20 @@ private void initPolicesCache(SystemTopicClient.Reader<PulsarEvent> reader, Comp
                 policyCacheInitMap.computeIfPresent(
                         reader.getSystemTopic().getTopicName().getNamespaceObject(), (k, v) -> true);
                 // replay policy message
-                policiesCache.forEach(((topicName, topicPolicies) -> {
-                    if (listeners.get(topicName) != null) {
-                        for (TopicPolicyListener<TopicPolicies> listener : listeners.get(topicName)) {
-                            listener.onUpdate(topicPolicies);
+                try {
+                    policiesCache.forEach(((topicName, topicPolicies) -> {
+                        if (listeners.get(topicName) != null) {
+                            for (TopicPolicyListener<TopicPolicies> listener : listeners.get(topicName)) {
+                                listener.onUpdate(topicPolicies);
```

Comment:
Can we easily catch the `onUpdate` method's exception with the warn log and then normally complete the future? 

IMO, Because it is an implementation problem(bug) in the listener logic, we can't let one bad apple spoils the whole bunch. :)

Reply:
If we complete directly, it may introduce data inconsistency？ Complete directly or throws exception depends on the importance of this method. From the source pr(https://github.com/apache/pulsar/pull/11136) of this code, I think this method is very important.

Reply:
If the listener throws an exception, it should be a bug. Do you have any ability to auto-fix it by retry?

Reply:
We need to fix this bug instead of ignoring it.